src/lib/stepper/step-header.ts

```diff
@@ -25,12 +33,15 @@ import {MatStepLabel} from './step-label';
   preserveWhitespaces: false,
 })
 export class MatStepHeader implements OnDestroy {
-  /** Icon for the given step. */
-  @Input() icon: string;
+  /** State of the given step. */
+  @Input() state: string;
 
   /** Label of the given step. */
   @Input() label: MatStepLabel | string;
 
+  /** Overrides for the header icons, passed in via the stepper. */
+  @Input() iconOverrides: {[key: string]: TemplateRef<any>};
```

Comment:
Why not a map?

Reply:
It doesn't need to be. It's just a `string->TemplateRef` map.